Testscripts/Linux/dpdkPmdsBasicCheck.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+# This script will run DPDk basic testpmd command in different PMDs and check failures.
+# To run this script constants.sh details must.
+#
+########################################################################################################
+
+HOMEDIR=$(pwd)
+LOGDIR="${HOMEDIR}/DpdkTestPmdLogs"
+CONSTANTS_FILE="./constants.sh"
+UTIL_FILE="./utils.sh"
+DPDK_UTIL_FILE="./dpdkUtils.sh"
+FAILSAFE_FILE="./failsafe.sh"
+
+. ${CONSTANTS_FILE} || {
+	echo "ERROR: unable to source ${CONSTANTS_FILE}!"
+	echo "TestAborted" > state.txt
+	exit 1
+}
+. ${UTIL_FILE} || {
+	echo "ERROR: unable to source ${UTIL_FILE}!"
+	echo "TestAborted" > state.txt
+	exit 2
+}
+. ${DPDK_UTIL_FILE} || {
+	echo "ERROR: unable to source ${DPDK_UTIL_FILE}!"
+	echo "TestAborted" > state.txt
+	exit 2
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+LogMsg "*********INFO: Script execution Started********"
+clientIPs=($(hostname -I))
+if [[ ${clientIPs[1]} == "" ]] || [[ ${clientIPs[2]} == "" ]];
+then
+	LogMsg "Extra NICs doesn't have ips, do dhclient"
+	dhclient eth1 eth2
+	sleep 5
+	clientIPs=($(hostname -I))
+else
+	LogMsg "Extra NICs have ips, collect them for test."
+fi
+echo -e "client=${clientIPs[0]}\nserver=${clientIPs[0]}" >> ${CONSTANTS_FILE}
+echo -e "clientNIC1ip=${clientIPs[1]}\nclientNIC2ip=${clientIPs[2]}" >> ${CONSTANTS_FILE}
+. ${CONSTANTS_FILE}
+echo "client-vm : eth0 : ${client} : eth1 : ${clientNIC1ip} eth2 : ${clientNIC2ip}"
+
+function checkCmdExitStatus ()
+{
+	exit_status=$?
```

Comment:
Does it work? I think you need to pass $? as a parameter.

Reply:
Thanks! I double check that function and I think it can work well. No need to pass $? as a parameter.